test/docker-e2e/e2e_upgrade_test.go

```diff
@@ -0,0 +1,199 @@
+			"--yes",
+		}
+		stdout, stderr, err := node.Exec(ctx, signalCmd, nil)
+		s.Require().NoError(err, "failed to signal for upgrade: %s", stderr)
+		s.T().Logf("Signal output: %s", stdout)
+	}
+
+	s.Require().NoError(wait.ForBlocks(ctx, 2, chain))
+
+	hostname, err := validatorNode.GetInternalHostName(ctx)
+	s.Require().NoError(err, "failed to get internal hostname")
+
+	// Query the tally to see if we have enough voting power
+	tallyCmd := []string{
+		"celestia-appd", "query", "signal", "tally", fmt.Sprintf("%d", targetAppVer),
+		"--node", fmt.Sprintf("tcp://%s:26657", hostname),
+		"--output", "json",
+	}
+	tallyStdout, tallyStderr, err := validatorNode.Exec(ctx, tallyCmd, nil)
+	s.Require().NoError(err, "failed to query tally: %s", tallyStderr)
+	s.T().Logf("Tally output: %s", tallyStdout)
+
+	// Execute try-upgrade transaction on all nodes
+	for i, node := range chain.GetNodes() {
+		hostname, err := node.GetInternalHostName(ctx)
+		s.Require().NoError(err, "failed to get internal hostname")
+
+		s.T().Logf("Executing try-upgrade transaction on validator %d", i)
+		tryUpgradeCmd := []string{
+			"celestia-appd", "tx", "signal", "try-upgrade",
+			"--home", homeDir,
+			"--from", records[i].Name,
+			"--keyring-backend", "test",
+			"--chain-id", chainID,
+			"--fees", "200000utia",
+			"--node", fmt.Sprintf("tcp://%s:26657", hostname),
+			"--yes",
+		}
+		upgradeStdout, upgradeStderr, err := node.Exec(ctx, tryUpgradeCmd, nil)
+		s.Require().NoError(err, "failed to execute try-upgrade on validator %d: %s", i, upgradeStderr)
+		s.T().Logf("Try-upgrade output from validator %d: %s", i, upgradeStdout)
+	}
+
+	s.T().Log("Querying upgrade info")
+	upgradeInfoCmd := []string{
+		"celestia-appd", "query", "signal", "upgrade",
+		"--node", fmt.Sprintf("tcp://%s:26657", hostname),
+		"--output", "json",
+	}
+	upgradeInfoStdout, upgradeInfoStderr, err := validatorNode.Exec(ctx, upgradeInfoCmd, nil)
+	s.Require().NoError(err, "failed to query upgrade info: %s", upgradeInfoStderr)
+	s.T().Logf("Upgrade info: %s", upgradeInfoStdout)
+
+	// Wait for the upgrade to be scheduled
+	s.Require().NoError(wait.ForBlocks(ctx, 5, chain))
+
+	err = chain.Stop(ctx)
+	s.Require().NoError(err)
+
+	chain.UpgradeVersion(ctx, targetVersion)
+
+	err = chain.Start(ctx)
+	s.Require().NoError(err)
+
+	err = wait.ForBlocks(ctx, 2, chain)
+	s.Require().NoError(err)
+
+	// Verify the version after upgrade
+	rpcClient, err = validatorNode.GetRPCClient()
+	s.Require().NoError(err, "failed to get RPC client for version check")
+
+	abciInfo, err = rpcClient.ABCIInfo(ctx)
+	s.Require().NoError(err, "failed to fetch ABCI info")
+
+	// The version string might vary, but should contain the PR number
+	versionStr := abciInfo.Response.GetVersion()
+	s.Require().True(strings.Contains(versionStr, "d13d38a"), "version should contain PR number")
+
+	// Verify app version is upgraded
+	s.Require().Equal(targetAppVer, abciInfo.Response.GetAppVersion(), "app_version mismatch")
```

Comment:
it might be nice to do some basic assertion that you can still send transactions before and after the upgrade, [this function](https://github.com/celestiaorg/celestia-app/blob/100f351cdb01e291b4ff1652f3326b23ff3b719b/test/docker-e2e/e2e_simple_test.go#L93) exists as a basic sanity check you can probably use as-is